ts/sagas/wallet.ts

```diff
@@ -343,6 +345,62 @@ function* showWalletOrSelectPsp(idWallet: number, paymentId?: string) {
   }
 }
 
+const MAX_RETRIES_POLLING = 20;
+const DELAY_BETWEEN_RETRIES_MS = 1000;
+
+const delay = (ms: number): Promise<undefined> =>
+  new Promise(resolve => setTimeout(resolve, ms));
+
+const pollForPaymentId = async (
+  backendClient: ReturnType<typeof BackendClient>,
+  p: { paymentContextCode: CodiceContestoPagamento },
+  retries = MAX_RETRIES_POLLING
+): Promise<string | undefined> => {
+  if (retries === 0) {
+    return undefined;
+  }
+  const response = await backendClient.getPaymentId(p);
+  if (response === undefined) {
+    return undefined;
+  }
+  if (response.status === 200) {
+    return response.value.idPagamento;
+  }
+  if (response.status === 404) {
+    await delay(DELAY_BETWEEN_RETRIES_MS);
+    console.warn("waiting");
+    return pollForPaymentId(backendClient, p, retries - 1);
+  }
+  return undefined;
+};
+
+// WIP: is this the appropriate place for this function?
+const attivaRpt = async (
+  sessionToken: SessionToken | undefined,
+  rptId: RptId,
+  paymentContextCode: CodiceContestoPagamento,
+  amount: AmountInEuroCents
+) => {
+  if (sessionToken) {
+    const backendClient = BackendClient(apiUrlPrefix, sessionToken);
+
+    const response:
+      | BasicResponseTypeWith401<PaymentActivationsPostResponse>
+      | undefined = await backendClient.postAttivaRpt({
+      rptId,
+      paymentContextCode,
+      amount
+    });
+    console.warn(response);
+    if (response !== undefined && response.status === 200) {
+      // successfully request the payment activation
+      // now poll until a paymentId is made available
+      return await pollForPaymentId(backendClient, { paymentContextCode });
```

Comment:
I would split the two "stages", `attivaRpt` will just do the _attiva_ and return a successful `Promise`, the the caller can call `pollForPaymentId` and wait